### PR TITLE
feat: returns error on invalid proof range in buildRangeProof

### LINKED ISSUE #142: buildRangeProof can panic upon invalid input range
# Problem
The [buildRangeProof](https://github.com/celestiaorg/nmt/blob/master/nmt.go#L244) can panic if the `start` and `end` arguments are invalid. 

# Acceptance Criteria
Add an initial check for the input range, and make an early return if invalid.

### PR DESCRIPTION
**Pull request by staheri14**
## Overview

Closes #142 

## Checklist
- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by cmwaters — DISMISSED** [review]
LGTM. I'm assuming that the range includes `start` and excludes `end` right?

**Review by evan-forbes — DISMISSED** [review]
+1 to Callum's comment, otherwise LGTM!

**Comment by staheri14** [comment]
> LGTM. I'm assuming that the range includes start and excludes end right?

Yes, you are right, I added a comment to the function to make it more clear, please see https://github.com/celestiaorg/nmt/pull/182/commits/fcd3cd0b120447b85a002935dda0e2acc3530f93

**Review by cmwaters — APPROVED** [review]

File: nmt.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func (n *NamespacedMerkleTree) ProveRange(start, end int) (Proof, error) {
 	isMaxNsIgnored := n.treeHasher.IsMaxNamespaceIDIgnored()
 	// TODO: store nodes and re-use the hashes instead recomputing parts of the
 	// tree here
-	if start < 0 || start >= end || end > len(n.leafHashes) {
+	if err := n.validateRange(start, end); err != nil {
 		return NewEmptyRangeProof(isMaxNsIgnored), ErrInvalidRange
```
Comment (rootulp) [inline nmt.go:174]: Per the other suggestion, should this be

```suggestion
		return NewEmptyRangeProof(isMaxNsIgnored), err
```

**Review by rootulp — APPROVED** [review]

File: nmt.go (hunk 1 above)
Comment (staheri14) [reply nmt.go]: Correct, done https://github.com/celestiaorg/nmt/pull/182/commits/d491b529c5e6d7544b126d4879d52a14c785dd69

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**